src/proto/shim/mod.rs

```diff
@@ -19,7 +50,15 @@ use core::convert::TryInto;
 #[unsafe_guid("605dab50-e046-4300-abb6-3dd810dd8b23")]
 #[derive(Protocol)]
 pub struct ShimLock {
-    verify: extern "efiapi" fn(buffer: *const u8, size: u32) -> Status,
+    verify: extern "sysv64" fn(buffer: *const u8, size: u32) -> Status,
+    hash: extern "sysv64" fn(
+        buffer: *const u8,
+        size: u32,
+        context: *mut Context,
+        sha256: *mut [u8; SHA256_DIGEST_SIZE],
+        sha1: *mut [u8; SHA1_DIGEST_SIZE],
```

Comment:
Is it correct for these two to pass in a pointer to a slice, rather than a `*mut u8`?

Reply:
I think it's better this way, makes it clear what the pointer refers to. Are these pointers required to be aligned, or will they work with any buffer?

Reply:
> Is it correct for these two to pass in a pointer to an array, rather than a `*mut u8`?

For `extern "C"` FFI these are the same thing, and it makes the code less error prone. 

> Are these pointers required to be aligned, or will they work with any buffer?

Given the usage in MokManger/Shim and the protocol's declaration, no alignment is required.